### DISCUSSION TITLE
GITHUB PRIVATE OCTERNSHIP ERROR ON TESTING FUNCTION !

### DISCUSSION THREAD

**Discussion by ManoBharathi93** [GitHub-Octernships/Ivy-Octernships-ML#9] [Q&A]
Did anyone who applied for github octernship have completed the task with 100 % test passed if so kindly reply to msg so that we get to know whether issue is from our side or not !

In github assigned private repo
- [ ] Does your implementation passes 100%  of test in local development ?
- [ ] Does Your PR passes 100 % of test in GitHub CI/CD  test coverage ?

If yes, then kindly shows us How do you setup the repo in local development.

**Comment by ManoBharathi93** [comment]
If anyone facing the same issue kindly acknowledge here...

**Comment by SumantxD** [comment]
@ManoBharathi93 Gald someone asked, I am facing similar issue, After i solved a ToDo list task i sent a PR to the public ivy repo, all the tests passed in the GitHub CI/CD, but when i send PR to private octernship repo none of the tests passes, i think it is due to the fact that , running CI/CD tests in a private repo will require spending limit  to be increased under 'Billing & plans' which will require github pro membership , Really confused what to do :disappointed: 
I don't know if my application will be considered or not this way, should i like the PR in the description of the private PR , so that the team can see that the code works fine?
Or i can submit it the way it is?
PLZ respond if you are able to figure out any of these.

**Comment by bilal-aamer** [comment]
Well said @SumantxD, I am facing the same problem, however, how i understand it, our private repo as part of our GitHub Classroom assignment doesnt have the billing plan as in Ivy's public repo so thats why the said "Billings & plans" note.

Regarding the procedure to sumbit the private PR, looking for a review.

**Comment by bit-aloo** [comment]
Hello Everyone, I am getting this specific error while running `pytest ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_manipulation.py::function_name`. But the same works well with actual repository. If anyone can point out what might be the error. I have setup codebase via codespace.
<details close>
<summary>Error Image</summary>
<img width="1326" alt="Screenshot 2023-08-13 at 6 50 35 PM" src="https://github.com/GitHub-Octernships/Ivy-Octernships-ML/assets/84662239/61bce7a2-7043-4072-809a-eeb0e43fe36f">
</details>

**Comment by ManoBharathi93** [comment]
This error is not caused by your Implementation. This is common error which occurs mainly on the private repo.

**Reply by bit-aloo** [reply]
Any idea how to overcome this and test the implementation. Would be a great help!!!

**Reply by ManoBharathi93** [reply]
try to pass all the test using [ivy main repo ](https://github.com/unifyai/ivy). then push  that code in GitHub assigned private repo and merge it. this is what i done so far.

**Reply by bilal-aamer** [reply]
Any vote of confirmation or review by the Ivy staff yet @ManoBharathi93 for your private merged PR?

**Reply by ManoBharathi93** [reply]
KINDLY CHECK THE IVY DISCORD -> GENERAL SECTION.

**Comment by AlexandraPosta** [comment]
Hey guys! 

There is no issue with your local setups. As mentioned [here](https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/15), it is an integration problem on our side. It is not related to the payment method, just test integration and out-of-date version of repositories.

**Reply by Aarsh2001** [reply]
Just a follow up on this, we'll soon implement CI testing on your repositories as well 🙂

**Comment by Shivam7-1** [comment]
Hii  I am new to Octernships program
I am unable to apply for IVY Octernships Program getting Error Like this
" Oh oh, looks like you need to contribute a bit more to the GitHub community before you’re eligible for an Octernship. Check out our guide on how to contribute to open-source projects to learn how you can make a meaningful impact! Keep up the great work and you’ll be a perfect fit in no time! "
Can Anyone Tell me how much Should I Contribute to GitHub for Become Eligible For Octernship Program ?

**Reply by AlexandraPosta** [reply]
Please check here: https://github.com/GitHub-Octernships/Ivy-Octernships-ML/discussions/1#discussioncomment-6739452

**Comment by AKACHI-4** [comment]
I have a doubt that if two contributors contribute one single common frontend function so what will be happen ? like both of there contributions are accepted as octernship task or else ?? 

Please reply

**Reply by AlexandraPosta** [reply]
The contribution is made in your private Octernship repository, so technically you should not be able to see other implementations. As long as you reference one of the open tasks (green dot rather then purple) your implementation will be taken into consideration (please do not open PRs for hflip/vflip/sort/add_ , implementations for these functions will only be considered if the PR was made before the functions were merged in our codebase).
